src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -786,7 +786,7 @@ public BaseResponseBean advanceSessionWithEndpoint(MenuSession menuSession, Stri
         // Sync requests aren't run when executing operations, so stop and check for them after each operation
         for (StackOperation op : endpoint.getStackOperations()) {
             sessionWrapper.executeStackOperations(new Vector<>(Arrays.asList(op)), evalContext);
-            Screen screen = menuSession.getNextScreen(false, new EntityScreenContext());
+            Screen screen = menuSession.getNextScreenIfSyncScreen(false, new EntityScreenContext());
             if (screen instanceof FormplayerSyncScreen) {
```

Comment:
seems like this if statement isn't necessary now

Reply:
an if statement is still needed, but it can be updated to `if (screen != null)`

Reply:
[6d17f14](https://github.com/dimagi/formplayer/pull/1608/commits/6d17f14eac50f83d418be65ec0e56222744e01de)